src/joint_state_listener.cpp

```diff
@@ -92,15 +92,19 @@ JointStateListener::~JointStateListener()
 
 std::string JointStateListener::getTFPrefix()
 {
+  if (tf_prefix_cached_) {
+    return tf_prefix_;
+  }
+
   ros::NodeHandle n_tilde("~");
-  std::string tf_prefix;
 
   // get the tf_prefix parameter from the closest namespace
   std::string tf_prefix_key;
   n_tilde.searchParam("tf_prefix", tf_prefix_key);
-  n_tilde.param(tf_prefix_key, tf_prefix, std::string(""));
+  n_tilde.param(tf_prefix_key, tf_prefix_, std::string(""));
```

Comment:
Do you want to use `getCached`?

Reply:
Is there a cached version of `searchParam`?

Reply:
@peroxyacyl I don't think that there is a cached version of `searchParam`. The current implementation looks good to me.

Reply:
> Is there a cached version of searchParam?

No, now LGTM without any further change!